src/modules/nextcloud/settings.js

```diff
@@ -0,0 +1,150 @@
+import React from "react";
+import ConfigurationForm from "../../components/configuration-form";
+import { Button, Callout, FormGroup } from "@blueprintjs/core";
+import { useStateLink } from "@hookstate/core";
+import { remote } from "electron";
+import { notify } from "../../services/notification-service";
+import log from "electron-log";
+import { useQuery } from "react-query";
+
+async function signIn(configuration) {
+  const { url } = configuration.get();
+
+  const res = await fetch(url + "/index.php/login/v2", {
+    credentials: "omit",
+    method: "POST",
+    headers: {
+      Content: "application/json",
+      Origin: url,
+    },
+  });
+
+  const authInfos = await res.json();
+
+  const authWindow = new remote.BrowserWindow({ width: 600, height: 800, show: true });
+
+  let authDisplayed = true;
+  let authenticated = false;
+
+  authWindow.on("closed", () => {
+    authDisplayed = false;
+  });
+
+  authWindow.loadURL(authInfos.login);
+
+  const { token, endpoint } = authInfos.poll;
+
+  while (authDisplayed && !authenticated) {
```

Comment:
I don't think you need a loop here. You could just return a promise no?

Reply:
No, we need to poll the URL until the user login, see https://docs.nextcloud.com/server/18/developer_manual/client_apis/LoginFlow/index.html#login-flow-v2
To limit the number of request, I add a timeout of 1 second before retry.